torchrl/envs/transforms/transforms.py

```diff
@@ -1172,10 +1173,18 @@ def _apply_transform(
         self, reward: torch.Tensor, target_return: torch.Tensor
     ) -> torch.Tensor:
         if self.mode == "reduce":
-            target_return = target_return - reward
+            if reward.ndim == 1 and target_return.ndim == 2:
+                # if target is stacked
+                target_return = target_return[-1] - reward
+            else:
+                target_return = target_return - reward
             return target_return
         elif self.mode == "constant":
-            return target_return
+            if reward.ndim == 1 and target_return.ndim == 2:
+                # if target is stacked
+                target_return = target_return[-1] - reward
+            else:
+                target_return = target_return - reward
```

Comment:
Can you comment on this?
What's the problem we're trying to solve?
Should be in a separate PR I believe

Reply:
target_return in case of the DT has shape (20,1) including context of the past 19 target_returns.

```
if reward.ndim == 1 and target_return.ndim == 2:
                # if target is stacked
                target_return = target_return[-1] - reward
            else:
                target_return = target_return - reward
```

This just checks if it is the case and we have context. If it is the case we only want to reduce the current target return of shape (20,1) thats why `target_return[-1] - reward`.
    
For the constant case Ill change it as we here dont need to reduce and the if/else is also not needed to check so we return the target_return.